### PR TITLE
Virginia Adjusted Gross Income (VAGI)

### LINKED ISSUE #2466: Virginia Adjusted Gross Income (VAGI)
FAGI
+addition
-age deduction
-subtractions

### PR DESCRIPTION
**Pull request by oliverh233**
Fixes #2466

### CODE REVIEW

File: policyengine_us/variables/gov/states/va/tax/income/additions/va_interest_on_obligations_of_other_states.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class va_interest_on_obligations_of_other_states(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Virginia interest on obligations of other states"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/",
+        "§ 58.1-322.01.(1.)",
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/additions/va_interest_on_obligations_of_other_states.py:12]:
```suggestion
        # § 58.1-322.01.(1.) 
```

File: policyengine_us/variables/gov/states/va/tax/income/additions/va_interest_on_obligations_of_other_states.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/additions/va_interest_on_obligations_of_other_states.py:1]: dont need this

File: policyengine_us/variables/gov/states/va/tax/income/additions/va_other_additions.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/additions/va_other_additions.py:1]: we can just make one va_additions.py for all additions

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_real_estate_investment_trust_subtraction.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+class va_real_estate_investment_trust_subtraction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Virginia Real Estate Investment Trust subtraction"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/",
+        "§ 58.1-322.02.(28.a.)",
```
Comment (Mshuning) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_real_estate_investment_trust_subtraction.py:12]:
```suggestion
    reference = (
        title="§ 58.1-322.02.(28.a.)",
        href="https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/",
```
@PavelMakarchuk is it ok to write reference like this?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/subtractions/va_real_estate_investment_trust_subtraction.py:12]: yes that works well here

File: policyengine_us/variables/gov/states/va/tax/income/va_agi.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
+    )
+    defined_for = StateCode.VA
+
+    adds = [
+        "adjusted_gross_income",
+        "va_additions",
+    ]
+    subtracts = [
+        "va_military_basic_pay_subtraction",
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_agi.py:20]: make a va_subtractions.py and a list parameter that includes all subtarctions

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_agi.py:20]: also need us_govt_interest, from point 1 (law) if I am not wrong

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_agi.py:20]: double check all [subtarctions](https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/)

**Comment by PavelMakarchuk** [comment]
duplicate PR

**Pull request closed without merging**